### PR TITLE
fix: storybook in prod

### PR DESCRIPTION
**Pull request by Barsnes**
args tables are not currently working in production. 
This PR fixes this. `react-docgen` is still used in local development, due to #2173, so you have to check this is production.

See preview deployment.

Also updates to a minor version of storybook.

### CODE REVIEW

**Comment by mimarz** [comment]
This can also be due to where Storybook config files are located in relative to the source code its supposed to "read" for infering props table or whats imported and used in storybook.

Try reverting the package.json and tsconfig and just refer to `src` instead of "built" code.

We had the same issue in Equinor Design System.

File: apps/storybook/.storybook/main.ts [hunk 1]
```diff
@@ -35,7 +34,6 @@ const config: StorybookConfig = {
     getAbsolutePath('@storybook/addon-links'),
     getAbsolutePath('@storybook/addon-essentials'),
     getAbsolutePath('@storybook/addon-interactions'),
-    getAbsolutePath('@storybook/addon-storysource'),
```
Comment (mimarz) [inline apps/storybook/.storybook/main.ts:38]: Was this supposed to be removed?

Comment (Barsnes) [reply apps/storybook/.storybook/main.ts:38]: No it was not :) Fixing

Comment (Barsnes) [reply apps/storybook/.storybook/main.ts:38]: Added back in latest commit

**Pull request merged**